src/python/pants/backend/codegen/tasks/wire_gen.py

```diff
@@ -42,7 +40,23 @@ def register_options(cls, register):
   def __init__(self, *args, **kwargs):
     """Generates Java files from .proto files using the Wire protobuf compiler."""
     super(WireGen, self).__init__(*args, **kwargs)
-    self.java_out = os.path.join(self.workdir, 'gen-java')
+
+  @property
+  def synthetic_target_type(self):
+    return JavaLibrary
+
+  def is_gentarget(self, target):
+    return isinstance(target, JavaWireLibrary)
+
+  def sources_generated_by_target(self, target):
```

Comment:
Is there an easy way to construct a dummy target for that, outside of the build graph? If not, the integration tests should cover this implicitly.